packages/kernel/src/socket.ts

```diff
@@ -0,0 +1,43 @@
+import { KernelMessage } from '@jupyterlab/services';
+
+import { WebSocket } from 'mock-socket';
+
+// @ts-ignore: No type definitions for this
+import { createMessageEvent } from 'mock-socket/src/event/factory';
+
+// @ts-ignore: No type definitions for this
+import networkBridge from 'mock-socket/src/network-bridge';
+
+// @ts-ignore: No type definitions for this
+import delay from 'mock-socket/src/helpers/delay';
+
+
+/**
+ * A fake socket that doesn't require serialization/deserialization of messages
+ */
+export class KernelSocket extends WebSocket {
+
+    /*
+     * Send an IMessage through the socket
+     */
+    sendMessage (data: KernelMessage.IMessage) {
```

Comment:
This is more or less a copy of `WebSocket.send` only that it doesn't force the `data` to be a string

Reply:
Instead it passes the `KernelMessage.IMessage` without touching it